message/validation/validation.go

```diff
@@ -229,6 +237,11 @@ func (mv *messageValidator) ValidatorForTopic(_ string) func(ctx context.Context
 // ValidatePubsubMessage validates the given pubsub message.
 // Depending on the outcome, it will return one of the pubsub validation results (Accept, Ignore, or Reject).
 func (mv *messageValidator) ValidatePubsubMessage(_ context.Context, peerID peer.ID, pmsg *pubsub.Message) pubsub.ValidationResult {
+	if mv.peerRateLimiter != nil && !mv.peerRateLimiter.AllowRequest(peerID) {
```

Comment:
I'd suggest instead of allowing peerRateLimiter to be nil, to either always have to pass it (even in tests), or make it an interface and build an additional "no-op" implementation used for tests only.